src/view/components/button.rs

```diff
@@ -1,33 +1,75 @@
 use crate::view::{
   handler::MouseState,
-  renderer::{RenderCtx, ViewResult},
+  renderer::{Component, RenderCtx, ViewResult},
 };
 use sdl2::{pixels::Color, rect::Rect};
 
-pub fn button<'renderer, H: 'renderer>(
+#[derive(PartialEq)]
+pub struct ButtonProps {
+  pub border_color: Color,
+  pub color_on_hover: Color,
+  pub mouse: MouseState,
+}
+
+pub struct Button<H> {
+  props: ButtonProps,
   bounds: Rect,
-  border_color: Color,
-  color_on_hover: Color,
-  mut on_click: H,
-) -> impl FnMut(RenderCtx, &MouseState) -> ViewResult + 'renderer
+  on_click: H,
+}
+
+impl<H> Button<H>
+where
+  H: FnMut(),
```

Comment:
```suggestion
impl<H: FnMut()> Button<H>
```
これぐらいだったら素直にこれでいい気がする

Reply:
dd1a1a2 で対応